util/job_manager.rb

```diff
@@ -58,22 +58,22 @@ def perform
         @wrapper = wrapper
         log "avancement: #{message} #{avancement}"
         if avancement && cost
-          p = Result.get(self.uuid) || { 'graph' => [] }
-          p['graph'] = [] if !p.has_key?('graph')
-          p['csv'] = true if ask_restitution_csv
-          p['graph'] << { iteration: avancement, cost: cost, time: time }
-          p['result'] = solution if solution
+          p = Result.get(self.uuid) || { graph: [] }
+          p[:graph] ||= []
+          p[:csv] = true if ask_restitution_csv
+          p[:graph] << { iteration: avancement, cost: cost, time: time }
+          p[:result] = solution
```

Comment:
The solution is not always transmitted if we have an intermediate cost. We should not override the result in this case

Reply:
My mistake :) I thought p[:result] could not be different than nil if solution was nil.